README.rst

```diff
@@ -779,6 +779,76 @@ values of this attribute, ordered in priority from high to low. Using this optio
 label. In this example an RQT item with multiple TEST items, one with a *fail* and others a *pass* as *result* value in
 the TEST_REP item, will be added to the *fail* slice of the pie chart.
 
+
+Retrieving checklist values from GitLab/GitHub (advanced)
+=========================================================
+
+The plugin can add an additional attribute to a traceability item if its item ID exists in a checklist inside the
+description of a merge/pull request. Documentation items can be linked to a checklist by defining them with the
+*checklist-item* directive.
+
+.. code-block:: rest
+
+    .. checklist-item:: PLAN-UNIT_TESTS Have you added unit tests for regression detection?
+
+This directive gets treated the same as the *item* directive, except that an additional attribute gets added when the
+item's ID exists in the checklist. The attribute's value depends on the status of the checkbox. The plugin queries the
+API of GitLab/GitHub for the description of the configured merge request.
+
+The configuration of this feature is stored in the configuration variable *traceability_checklist*.
+
+.. code-block:: python
+
+    traceability_checklist = {
+        'private_token': config('PRIVATE_TOKEN'),
+        'api_host_name': config('API_HOST_NAME'),
+        'project_id': config('PROJECT_ID'),
+        'merge_request_id': config('MERGE_REQUEST_ID'),
+        'attribute_name': config('TARGET_ATTRIBUTE_NAME'),
+        'attribute_to_str': config('TARGET_ATTRIBUTE_TO_STRING'),
+        'attribute_values': config('TARGET_ATTRIBUTE_VALUES'),
+    }
+
+In this default configuration the variables are fetched from the environment, e.g. by using a ``.env`` file.
+
+.. code-block:: bash
+
+    # copy example .env to your .env
+    cp doc/.env.example .env
+
+    # add env variables by adjusting the template values in .env
+
+- *ATTRIBUTE_NAME* is the identifier of the attribute to be added, e.g. *checked*.
+- *ATTRIBUTE_TO_STRING* is the string representation (as to be rendered in html) of the attribute name, e.g. *Answer*.
+- *ATTRIBUTE_VALUES* are two comma-separated attribute values, e.g. *yes/no*. The first value is used when the checkbox is checked and the second value when unchecked.
+
+A query is sent to the GitLab/GitHub API to retrieve the status of every checkbox in the description of the merge/pull
+request. The traceability item's ID is expected to follow the checkbox directly.
+Example of a valid checklist in Markdown:
+
+.. code-block:: rest
+
+    - [x] PLAN-UNIT_TESTS Have you added unit tests for regression detection?
+    - [ ] PLAN-PACKAGE_TEST Have you tested the package?
+
+GitLab
+------
+
+- *PRIVATE_TOKEN* is your personal access token that has API access.
+- *API_HOST_NAME* is the host name of the API, e.g. *https://gitlab.example.com/api/v4*.
```

Comment:
can we remove `api/v4` from contents of this varialbe. `api/v4` is actually something hardcoded in plugin as that is the version of API we will support. We can move higher or lower, but I would not like to have this dynamically inputted by user.

Reply:
Although on the other side if someone has different gitlab instance then maybe this is a preferred way to keep into a separate variable. So nevermind.